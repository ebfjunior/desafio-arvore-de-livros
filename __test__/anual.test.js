import React from 'react';

import Anual from '../src/components/anual';

describe('Componente Anual', () => {
  it('Deve renderizar adequadamente', () => {
    const periodos = {
      "2016" : {
        "10" : {"2016-10-12" : 1, "2016-10-23" : 38},
        "12" : {"2016-12-02" : 17}
      },
      "2017" : {
        "01" : {"2017-01-05" : 8}
      }
    };
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2016-09-03'} data_final={'2017-09-03'}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('Estado deve ser inicializado corretamente', () => {
    const periodos = {
      "2016" : {
        "10" : {"2016-10-12" : 1, "2016-10-23" : 38},
        "12" : {"2016-12-02" : 17}
      },
      "2017" : {
        "01" : {"2017-01-05" : 8}
      }
    };
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2016-09-03'} data_final={'2017-09-03'}/>);

    expect(wrapper.state().periodos).toEqual(periodos);
    expect(wrapper.state().data_inicial).toEqual('2016-09-03');
    expect(wrapper.state().data_final).toEqual('2017-09-03');
  });

});