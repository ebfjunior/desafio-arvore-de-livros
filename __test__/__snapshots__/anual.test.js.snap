// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Anual Deve renderizar adequadamente 1`] = `
<Anual
  data_final="2017-09-03"
  data_inicial="2016-09-03"
  periodos={
    Object {
      "2016": Object {
        "10": Object {
          "2016-10-12": 1,
          "2016-10-23": 38,
        },
        "12": Object {
          "2016-12-02": 17,
        },
      },
      "2017": Object {
        "01": Object {
          "2017-01-05": 8,
        },
      },
    }
  }
>
  <span>
    Hello, √Årvore de Livros
  </span>
</Anual>
`;
